packages/core/src/workflow/sleep.ts

```diff
@@ -0,0 +1,94 @@
+import type { StringValue } from 'ms';
+import ms from 'ms';
+import { EventConsumerResult } from '../events-consumer.js';
+import { WorkflowSuspension } from '../global.js';
+import type { WorkflowOrchestratorContext } from '../private.js';
+import { withResolvers } from '../util.js';
+
+export function createSleep(ctx: WorkflowOrchestratorContext) {
+  return async function sleepImpl(param: StringValue | Date): Promise<void> {
+    const { promise, resolve } = withResolvers<void>();
+    const correlationId = `timer_${ctx.generateUlid()}`;
```

Comment:
All other ULIDs are with a four letter prefix, should we stay consistent? Maybe `timr` or `tmer`, or `wait`?